pkg/server/server.go

```diff
@@ -669,6 +671,67 @@ func (s *Server) handleKnativeChannelCRDs(w http.ResponseWriter, r *http.Request
 	s.KnativeChannelCRDLister.HandleResources(w, r)
 }
 
+// buildCSPDirectives takes the content security policy configuration from the server and constructs
+// a complete set of directives for the Content-Security-Policy-Report-Only header.
+// The constructed directives will include the default sources and the supplied configuration.
+// buildCSPDirectives takes the content security policy configuration from the server and constructs
+// a complete set of directives for the Content-Security-Policy-Report-Only header.
+// The constructed directives will include the default sources and the supplied configuration.
+func (s *Server) buildCSPDirectives() ([]string, error) {
```

Comment:
nit, this seems like it belongs in main.go with other config parsing logic. IMO, our http request handler shouldn't be concerned with parsing configs.

Reply:
I would argue placing it into the `main.go` since the `buildCSPDirectives` method is using fields of the `Server` object. If anything we could move the `ParseContentSecurityPolicyConfig` to a more appropriate pkg like `pkg/serverutils/utils.go`